src/qibolab/instruments/emulator/backends/generic.py

```diff
@@ -0,0 +1,315 @@
+
+    Returns:
+        The quantum operator if multiply_coeff is True, or a tuple containing the coefficient and operator string otherwise.
+    """
+    coeff, s, op_qid_list = inst
+
+    # construct op_dict and op_connectors_dict check dictionaries (involves strings and string operations)
+    if op_dict is None:
+        op_dict_check = {
+            "b": {},
+            "bdag": {},
+            "O": {},
+            "X": {},
+            "Z01": {},
+            "sp01": {},
+        }
+        for k in op_qid_list:
+            op_dict_check["b"].update({k: f"op_b_{k}"})
+            op_dict_check["bdag"].update({k: f"op_bdag_{k}"})
+            op_dict_check["O"].update({k: f"op_O_{k}"})
+            op_dict_check["X"].update({k: f"op_X_{k}"})
+            op_dict_check["Z01"].update({k: f"op_Z01_{k}"})
+            op_dict_check["sp01"].update({k: f"op_sp01_{k}"})
+        op_dict = op_dict_check
+
+    if op_connectors_dict is None:
+        op_connectors_dict_check = OrderedDict(
+            [
+                ("^", lambda a, b: f"tensor({a},{b})"),
+                ("*", lambda a, b: f"times({a},{b})"),
+                ("+", lambda a, b: f"plus({a},{b})"),
+                ("-", lambda a, b: f"minus({a},{b})"),
+            ]
+        )
+        op_connectors_dict = op_connectors_dict_check
+
+    def process_op(op_list: list, connector_list: List[str], connector: str) -> list:
+        """Implements connectors between operators."""
+        index_list = []
+        for i in range(connector_list.count(connector)):
+            ind = connector_list.index(connector)
+            index_list.append(ind + len(index_list))
+            connector_list.pop(ind)
+
+        new_op_list = []
+        for i, op in enumerate(op_list):
+            if i in index_list:
+                next_op = op_list[i + 1]
+                if i - 1 in index_list:
+                    new_op_list[-1] = op_connectors_dict[connector](
+                        new_op_list[-1], next_op
+                    )
+                else:
+                    new_op_list.append(op_connectors_dict[connector](op, next_op))
+            elif i - 1 in index_list:
+                pass
+            else:
+                new_op_list.append(op)
+        return new_op_list, connector_list
+
+    # convert operator instruction strings into list of components
+    split_inst = s.split(" ")
+    op_list = []
+    # implement operators
+    for op_key in split_inst[::2]:
+        op_name, qid = op_key.split("_")
+        op_list.append(op_dict[op_name][qid])
+    # implement operator connections
+    connector_list = split_inst[1::2]
+    for connector in list(op_connectors_dict.keys()):
+        op_list, connector_list = process_op(op_list, connector_list, connector)
+
+    if multiply_coeff:
+        return coeff * op_list[0]
+    else:
+        return coeff, op_list[0]
```

Comment:
I'm not sure why this should be an operation specific to the backend, and not a general emulator task.

Reply:
I'm confused, you are right that this is a general emulator task and that's why it is currently in `generic.py` instead of `qutip_backend.py`...

Reply:
I believe that the word "backend" here is potentially misleading (because of the Qibo backends). I'd suggest using something else, like "engine" or whatever you prefer.

Moreover, are you planning to implement further backends, other than QuTiP?

Reply:
I'm fine with "engine", and yes there are plans to implement additional engines other than QuTiP. Only QuTiP is included for now to basically set the base standard/reference through this PR for the others to follow.

Reply:
Renamed 'backend' to 'engine' in 0d5129a784f1ed495f971e27fa778fe4aab854c6 and 811d786432d6cfdcd012a718472e24b4c5954632.